astropy/visualization/tests/test_norm.py

```diff
@@ -265,6 +264,7 @@ def test_imshow_norm():
     import matplotlib.pyplot as plt
     image = np.random.randn(10, 10)
 
+    plt.clf()
```

Comment:
I am a bit surprised the initial `clf()` calls are needed here and in the other tests – would have expected each test is entered with a "clean" `plt` instance.

Reply:
Me too but the warning appears without this. 🤷

Reply:
The darker secrets of mpl artists... thanks for the fix!

Reply:
Imports are not going to be re-imported for every test; you will only get a clean instance if you explicitly reset it. You'll probably want to start using a pytest autouse fixture to do this for you automatically.

Reply:
Yes; I did not expect a fresh import (normally would have expected a single import at the top as in `test_misc.py` actually). more something like the fixture will do. An may have confused `pyplot` and `figure` instances in fact. 😝